src/useIsScrolled/useIsScrolled.js

```diff
@@ -0,0 +1,56 @@
+import { useRef, useEffect, useState } from "react";
+import ResizeObserver from "resize-observer-polyfill";
+
+import onScrollFn from "../utils/onScroll";
+import onResizeFn from "../utils/onResize";
+
+const useIsScrolled = () => {
```

Comment:
What bothers me is somehow repeated logic with the stateful component. It will be twice the work to maintain this lib. Can you think of any more ways to share some logic?